ChainSafe/gossamer
# feat(dot/telemetry): implement notify.finalized telemetry interface

## Changes



- Added NotifyFinalizedTM struct and send `notify.finalized` message when a block gets finalized
- Moved telemetry message structs to different files

## Tests

- Setup a local telemetry server using substrate-telemetry.

Use the branch `notify-finalized` from my fork `https://github.com/kishansagathiya/substrate-telemetry` (It print the message received, type of message and error. From what I have played around upstream doesn't do that.)

Run telementry backend (`telemetry_core` and `telemetry_shard`).

-  Run gossamer by setting telemetry  url to the local server. 

For this pull `kishansagathiya:issue-1502` (branch from https://github.com/ChainSafe/gossamer/pull/1890), since this feature is not merged at the time of writing.

Run `./bin/gossamer --chain polkadot --name 'Kishans Test Node' --telemetry-url 'ws://localhost:8001/submit 0'`

- Wait for message to get send and received. Follow logs of `telemetry_shard` command. After few minutes it should print output as follows
```
telemetry message received: {"best":"0xeb8822754b85d1bf5f6d7ad54b1244981944eaee01821d7d156c0574e61a6ab1","height":"254976","msg":"notify.finalized","ts":"2021-10-15T18:27:05.1594099+05:30"}
message type Payload::NotifyFinalized

```

```

```

## Issues



- Fixes #1835 

## Primary Reviewer 



- @edwardmack
